barman/clients/cloud_compression.py

```diff
@@ -0,0 +1,195 @@
+# -*- coding: utf-8 -*-
+# © Copyright EnterpriseDB UK Limited 2018-2021
+#
+# This file is part of Barman.
+#
+# Barman is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Barman is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Barman.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+The module handles compression specifics for barman-cloud so that compression
+algorithms requiring external dependencies can be lazily-imported and therefore
+treated as optional requirements.
+"""
+
+import bz2
+import gzip
+import shutil
+from abc import ABCMeta, abstractmethod
+from io import BytesIO
+
+from barman.utils import with_metaclass
+
+
+def _try_import_snappy():
```

Comment:
I agree - it's a bit of a pattern (anti-pattern, perhaps?) in barman-cloud. While it is intended to avoid users having to add dependencies which they will never use, such as the cloud providers, it does lead to that missing dependencies problem if someone isn't aware an optional dependency is required.

Having tested with snappy on the older distros like CentOS 7 and Debian 9 I'm actually less concerned now about adding this as a dependency - a simple "pip install python-snappy" does "just work" on those platforms, so it's not like it's a heavyweight dependency.

So I wonder if this is now overkill, and instead we make it a required dep and a top level import? What do you think?